src/Destructurama.Attributed/Attributed/AttributedDestructuringPolicy.cs

```diff
@@ -70,31 +70,31 @@ private CacheEntry CreateCacheEntry(Type type)
     {
         IPropertyDestructuringAttribute? GetPropertyDestructuringAttribute(PropertyInfo propertyInfo)
         {
-            var attr = propertyInfo.GetCustomAttributes().OfType<IPropertyDestructuringAttribute>().FirstOrDefault();
+            var attr = propertyInfo.GetCustomAttributesEx(_options.RespectMetadataTypeAttribute).OfType<IPropertyDestructuringAttribute>().FirstOrDefault();
             if (attr != null)
                 return attr;
 
             // Do not check attribute explicitly to not take dependency from Microsoft.Extensions.Telemetry.Abstractions package.
             // https://github.com/serilog/serilog/issues/1984
-            return _options.RespectLogPropertyIgnoreAttribute && propertyInfo.GetCustomAttributes().Any(a => a.GetType().FullName == "Microsoft.Extensions.Logging.LogPropertyIgnoreAttribute")
+            return _options.RespectLogPropertyIgnoreAttribute && propertyInfo.GetCustomAttributesEx(_options.RespectMetadataTypeAttribute).Any(a => a.GetType().FullName == "Microsoft.Extensions.Logging.LogPropertyIgnoreAttribute")
                 ? NotLoggedAttribute.Instance
                 : null;
         }
 
-        var classDestructurer = type.GetCustomAttributes().OfType<ITypeDestructuringAttribute>().FirstOrDefault();
+        var classDestructurer = type.GetCustomAttributesEx(_options.RespectMetadataTypeAttribute).OfType<ITypeDestructuringAttribute>().FirstOrDefault();
```

Comment:
This call will never return other attributes.

Reply:
Can't reproduce this, At least with Testcase AttributesAreConsultedWhenDestructuring it returns the class attribute.

Reply:
I mean it never returns **other** (new) attributes from metadata class because of `if` at the beginning of your method.